src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/mobile/set_global_properties_request.h

```diff
@@ -203,6 +203,29 @@ class SetGlobalPropertiesRequest
    */
   bool IsWhiteSpaceExist();
 
+  /**
+   * @brief helps to determine layout of interest. Returns keyboard layout,
+   * mentioned in current request. If not, returns saved keyboard layout for
+   * current app. If such layout wasn't saved, returns default keyboard layout
+   * (QWERTY)
+   * @return KeyboardLayout enum value
+   */
+  hmi_apis::Common_KeyboardLayout::eType GetKeyboardLayout();
+
+  /**
+   * @brief Returns allowed number of configurable keys for certain layout
+   * @return allowed number of configurable keys, if provided, and zero
+   * otherwise
+   */
+  uint32_t GetAllowedNumberOfConfigurableKeys();
+
+  /**
+   * @brief Checks if customizable keys are disallowed
+   * @return true if customizable keys are disallowed and false if allowed or
+   * this non-mandatoty parameter is absent or empty
+   */
+  bool IsCustomizableKeysDisallowed();
```

Comment:
@ychernysheva suggest to give preference to positive cases when evaluating some bool condition
```suggestion
  bool AreCustomizableKeysAllowed() const;
```

Reply:
@AKalinich-Luxoft But then we will need to check (!AreCustomizableKeysAllowed()), so maybe it will be simpler to check (AreCustomizableKeysDisallowed())?

Reply:
@ychernysheva one symbol is not a big deal, it will be simple anyway

Reply:
Done in f10bc21